### ISSUE TITLE
Terraform crash on awscc_kafkaconnect_connector

### ISSUE THREAD

**Issue by msfidelis** [hashicorp/terraform-provider-awscc#427]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

### Terraform CLI and Terraform AWS Cloud Control Provider Version

```
Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.2.0
+ provider registry.terraform.io/hashicorp/awscc v0.12.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
```
### Affected Resource(s)

* awscc_kafkaconnect_connector

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource awscc_kafkaconnect_connector "s3" {
    connector_name =  format("%s-s3-connect-%s", var.project_name, time_static.release.unix)
    service_execution_role_arn = aws_iam_role.msk.arn
    kafka_connect_version   = "2.7.1"
    kafka_cluster_encryption_in_transit = {
        encryption_type = "TLS"
    }
    kafka_cluster   = {
        apache_kafka_cluster = {
            bootstrap_servers = aws_msk_cluster.main.bootstrap_brokers_tls
            vpc = {
                subnets         = var.subnets
                security_groups = [
                    aws_security_group.sg.id
                ]
            }
        }
    }
    capacity = {
        provisioned_capacity = {
            worker_count = 1
            mcu_count    = 2
        }
    }
    kafka_cluster_client_authentication = {
        authentication_type = "NONE"
    }
    plugins = [
        {
            custom_plugin = {
                custom_plugin_arn   = aws_mskconnect_custom_plugin.s3.arn,
                revision            = aws_mskconnect_custom_plugin.s3.latest_revision
            }
        }
    ]
    connector_configuration = tomap({
    "connector.class"   =   "io.confluent.connect.s3.sink.S3SinkConnector",
    "format.class"      =   "io.confluent.connect.s3.format.bytearray.ByteArrayFormat"
    "s3.part.retries"   =   "3"
    "tasks.max"         =   "2",
    "s3.region"         =   "us-east-1",
    "timezone"          =   "America/Sao_paulo",
    "topics.dir"        =   "topics"
    "flush.size"        =   "1000"
    })
}
```

### Debug Output

Gist Debug: https://gist.github.com/msfidelis/477ea99bacb2452ff40381579a67e65f 

### Panic Output

Crash.log Output: https://gist.github.com/msfidelis/17d5320b431359244aba7cd87f44abd0

### Expected Behavior

The output should tell me what the error is with the data types and how to fix it.

### Actual Behavior

Terraform is crashing during plan and apply of resource `awscc_kafkaconnect_connector`

### Steps to Reproduce

1. `terraform plan`
2. `terraform apply`

### References

* #0000

**Comment by msfidelis** [comment]
I tried to upgrade most recent version and works fine  

```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.5.0
+ provider registry.terraform.io/hashicorp/awscc v0.14.0
```

**Comment by bflad** [comment]
I believe the source of the crash was fixed in Terraform CLI 1.0.3 with this change: https://github.com/hashicorp/terraform/pull/29207 (one of the contributing factors to why the terraform-plugin-framework documentation used to mention 1.0.3+ as the required version)

**Comment by ewbankkit** [comment]
```
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 391 [running]:
github.com/zclconf/go-cty/cty.Value.GetAttr(0x38b8778, 0xc00269ad50, 0x0, 0x0, 0xc0006b0770, 0x10, 0x38b8698, 0xc00005642f, 0x0, 0x0)
	/home/circleci/go/pkg/mod/github.com/zclconf/go-cty@v1.8.3/cty/value_ops.go:760 +0x416
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewAttributes(0xc0013b3260, 0x38b8778, 0xc0027e34f0, 0x305d2a0, 0xc0027f55c0, 0x38b8778, 0xc00269ad50, 0x0, 0x0, 0xc0027f56b0)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:267 +0x26f
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewNestedType(0xc00098a2e0, 0x38b8778, 0xc0027e34f0, 0x305d2a0, 0xc0027f55c0, 0x38b8778, 0xc00269ad50, 0x0, 0x0, 0x0, ...)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:314 +0x813
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewAttributes(0xc0013b2e10, 0x38b8778, 0xc0027e3510, 0x305d2a0, 0xc0027f5680, 0x38b8778, 0xc002893c30, 0x305d2a0, 0xc0028af9e0, 0xc0027f5350)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:295 +0x49a
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewNestedType(0xc00098a2c0, 0x38b8778, 0xc0027e3430, 0x0, 0x0, 0x38b8778, 0xc002893c30, 0x305d2a0, 0xc0028af9e0, 0xc0028af9e0, ...)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:314 +0x813
github.com/hashicorp/terraform/internal/plans/objchange.proposedNewAttributes(0xc0013b29f0, 0x38b8778, 0xc0027e3440, 0x305d2a0, 0xc0027f50b0, 0x38b8778, 0xc002893ef0, 0x305d2a0, 0xc0028afd10, 0x0)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:295 +0x49a
github.com/hashicorp/terraform/internal/plans/objchange.proposedNew(0xc0013b2d20, 0x38b8778, 0xc0027e3440, 0x305d2a0, 0xc0027f50b0, 0x38b8778, 0xc002893ef0, 0x305d2a0, 0xc0028afd10, 0x20, ...)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:80 +0x165
github.com/hashicorp/terraform/internal/plans/objchange.ProposedNew(0xc0013b2d20, 0x38b8778, 0xc002893970, 0x0, 0x0, 0x38b8778, 0xc002893ef0, 0x305d2a0, 0xc0028afd10, 0x38b8778, ...)
	/home/circleci/project/project/internal/plans/objchange/objchange.go:42 +0x11f
github.com/hashicorp/terraform/internal/terraform.(*NodeAbstractResourceInstance).plan(0xc001cb5340, 0x38ef9c0, 0xc0009a4ee0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/circleci/project/project/internal/terraform/node_resource_abstract_instance.go:726 +0xd7e
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).managedResourceExecute(0xc00209a720, 0x38ef9c0, 0xc0009a4ee0, 0xc000000001, 0xc00133acb8, 0xc00066dc50)
	/home/circleci/project/project/internal/terraform/node_resource_plan_instance.go:196 +0x8cb
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).Execute(0xc00209a720, 0x38ef9c0, 0xc0009a4ee0, 0xc00007e002, 0xc00066dce8, 0x100b805, 0x30a89c0)
	/home/circleci/project/project/internal/terraform/node_resource_plan_instance.go:54 +0xb3
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc002d0f110, 0x38ef9c0, 0xc0009a4ee0, 0x64531c8, 0xc00209a720, 0x0, 0x0, 0x0)
	/home/circleci/project/project/internal/terraform/graph_walk_context.go:129 +0xbf
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1(0x33fc440, 0xc00209a720, 0x0, 0x0, 0x0)
	/home/circleci/project/project/internal/terraform/graph.go:59 +0xbd3
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc0027c0ea0, 0x33fc440, 0xc00209a720, 0xc0027cc5c0)
	/home/circleci/project/project/internal/dag/walk.go:381 +0x288
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update
	/home/circleci/project/project/internal/dag/walk.go:304 +0x1246
```
